### PR TITLE
Dockerize build and test simple cluster replication

### PR DESCRIPTION
**Pull request by studzien**
Note that you need to run `make docker` before running `make test`. Building docker image is not test's dependency, so that the image is not built so often.

@pzel Can you check how long does `make test` take for you? It seems that the initial sync between two nodes requires ~20-30 seconds (at least on my machine).
This seems definitely too long for me.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+.PHONY: deps docker prepare rel test
 
 deps:
 	mix deps.get
 
+docker:
+	docker build . -t breaking-pp
+
+prepare:
+	mix local.hex --force
```
Comment (arkgil) [inline Makefile]: You also need `make local.rebar --force` here, otherwise `make rel` fails in when building Docker image.

Comment (studzien) [reply Makefile]: Hmm, does it crash for you? I believe we don't use any rebar dependency for now (cowboy uses erlang.mk)

Comment (arkgil) [reply Makefile]: Yup, I have an error when ranch is being compiled.

Comment (arkgil) [reply Makefile]: Which is strange, because it uses erlang.mk as well.

Comment (studzien) [reply Makefile]: Got it, it copied compiled beams from my local directory, so I also added a cleanup step, to make sure we compile everything from scratch in Docker.

Can you try now?

Comment (arkgil) [reply Makefile]: Works well now 👍

Comment (studzien) [reply Makefile]: Thanks! Good catch!

**Comment by arkgil** [comment]
@studzien on my machine `mix test` takes ~55s.

**Comment by pzel** [comment]
```
$ time make test
mix test 
.......

Finished in 9.8 seconds
7 tests, 0 failures

Randomized with seed 439026

real    0m10.502s
user    0m1.141s
sys     0m0.346s
```

**Pull request merged**